ci-operator/step-registry/gcp/provision/cco-manual-users/static/gcp-provision-cco-manual-users-static-commands.sh

```diff
@@ -0,0 +1,213 @@
+  for FILE in *; do
+    match_count=$(grep -c "${keyword}" "${FILE}")
+    if [ $match_count -ne 0 ]; then
+      echo "Remove CredentialsRequest '${FILE}' which is a '${keyword}' CR"
+      rm -f "${FILE}"
+    fi
+  done
+  popd
+}
+
+# Create credentials manifests file
+function create_credentials_manifests() {
+  local ns="$1"
+  local name="$2"
+  local b64_service_account_json="$3"
+  local target_dir="$4"
+  cat << EOF > ${target_dir}/99_${ns}_${name}-secret.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  namespace: ${ns}
+  name: ${name}
+data:
+  service_account.json: ${b64_service_account_json}
+EOF
+}
+
+# release-controller always expose RELEASE_IMAGE_LATEST when job configuraiton defines release:latest image
+echo "RELEASE_IMAGE_LATEST: ${RELEASE_IMAGE_LATEST:-}"
+# RELEASE_IMAGE_LATEST_FROM_BUILD_FARM is pointed to the same image as RELEASE_IMAGE_LATEST, 
+# but for some ci jobs triggerred by remote api, RELEASE_IMAGE_LATEST might be overridden with 
+# user specified image pullspec, to avoid auth error when accessing it, always use build farm 
+# registry pullspec.
+echo "RELEASE_IMAGE_LATEST_FROM_BUILD_FARM: ${RELEASE_IMAGE_LATEST_FROM_BUILD_FARM}"
+# seem like release-controller does not expose RELEASE_IMAGE_INITIAL, even job configuraiton defines 
+# release:initial image, once that, use 'oc get istag release:inital' to workaround it.
+echo "RELEASE_IMAGE_INITIAL: ${RELEASE_IMAGE_INITIAL:-}"
+if [[ -n ${RELEASE_IMAGE_INITIAL:-} ]]; then
+  tmp_release_image_initial=${RELEASE_IMAGE_INITIAL}
+  echo "Getting inital release image from RELEASE_IMAGE_INITIAL..."
+elif oc get istag "release:initial" -n ${NAMESPACE} &>/dev/null; then
+  tmp_release_image_initial=$(oc -n ${NAMESPACE} get istag "release:initial" -o jsonpath='{.tag.from.name}')
+  echo "Getting inital release image from build farm imagestream: ${tmp_release_image_initial}"
+fi
+# For some ci upgrade job (stable N -> nightly N+1), RELEASE_IMAGE_INITIAL and 
+# RELEASE_IMAGE_LATEST are pointed to different imgaes, RELEASE_IMAGE_INITIAL has 
+# higher priority than RELEASE_IMAGE_LATEST
+TESTING_RELEASE_IMAGE=""
+if [[ -n ${tmp_release_image_initial:-} ]]; then
+  TESTING_RELEASE_IMAGE=${tmp_release_image_initial}
+else
+  TESTING_RELEASE_IMAGE=${RELEASE_IMAGE_LATEST_FROM_BUILD_FARM}
+fi
+echo "TESTING_RELEASE_IMAGE: ${TESTING_RELEASE_IMAGE}"
+
+GOOGLE_PROJECT_ID="$(< ${CLUSTER_PROFILE_DIR}/openshift_gcp_project)"
+export GCP_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/gce.json"
+GCP_SERVICE_ACCOUNT=$(jq -r .client_email ${GCP_SHARED_CREDENTIALS_FILE})
+if ! gcloud auth list | grep -E "\*\s+${GCP_SERVICE_ACCOUNT}"
+then
+  gcloud auth activate-service-account --key-file="${GCP_SHARED_CREDENTIALS_FILE}"
+  gcloud config set project "${GOOGLE_PROJECT_ID}"
+fi
+
+# For OCP 4.15+
+CCO_CUSTOM_ROLE="projects/${GOOGLE_PROJECT_ID}/roles/installer_qe_cco_permissions"
+IMAGE_REGISTRY_CUSTOM_ROLE="projects/${GOOGLE_PROJECT_ID}/roles/installer_qe_image_registry_permissions"
```

Comment:
Suggest to add some comments to explain the included permission for the custom roles, and where the permission list gotten from.